libr/bin/format/mach0/mach0.c

```diff
@@ -1744,6 +1744,162 @@ static int inSymtab(SdbHt *hash, struct symbol_t *symbols, const char *name, ut6
 	return false;
 }
 
+void MACH0_(pull_symbols)(struct MACH0_(obj_t)* mo, RBinSymbolCallback cb, void *user) {
+	const char *symstr;
+	int j, s, stridx, symbols_size, symbols_count;
+	ut32 to, from, i;
+	//ut64 text_base = get_text_base (bin);
+
+	if (!mo || !mo->symtab || !mo->symstr) {
+		return;
+	}
+	/* parse symbol table */
+	/* parse dynamic symbol table */
+	symbols_count = (mo->dysymtab.nextdefsym + \
+			mo->dysymtab.nlocalsym + \
+			mo->dysymtab.nundefsym );
+	symbols_count += mo->nsymtab;
+	//symbols_count = mo->nsymtab;
+	symbols_size = (symbols_count + 1) * 2 * sizeof (struct symbol_t);
+
+	if (symbols_size < 1) {
+		return;
+	}
+	SdbHt *hash = sdb_ht_new ();
+	for (s = j = 0; s < 2; s++) {
+		switch (s) {
+		case 0:
+			from = mo->dysymtab.iextdefsym;
+			to = from + mo->dysymtab.nextdefsym;
+			break;
+		case 1:
+			from = mo->dysymtab.ilocalsym;
+			to = from + mo->dysymtab.nlocalsym;
+			break;
+#if NOT_USED
+		case 2:
+			from = mo->dysymtab.iundefsym;
+			to = from + mo->dysymtab.nundefsym;
+			break;
+#endif
+		}
+		if (from == to) {
+			continue;
+		}
+		from = R_MIN (R_MAX (0, from), symbols_size / sizeof (struct symbol_t));
+		to = R_MIN (to , symbols_size / sizeof (struct symbol_t));
+		to = R_MIN (to, mo->nsymtab);
+		int maxsymbols = symbols_size / sizeof (struct symbol_t);
+		if (to > 0x500000) {
+			eprintf ("WARNING: corrupted mach0 header: symbol table is too big %d\n", to);
+			sdb_ht_free (hash);
+			return;
+		}
+		if (symbols_count >= maxsymbols) {
+			symbols_count = maxsymbols - 1;
+		}
+		for (i = from; i < to && j < symbols_count; i++, j++) {
+			RBinSymbol *sym = R_NEW0 (RBinSymbol);
+			sym->paddr = addr_to_offset (mo, mo->symtab[i].n_value);
+			sym->vaddr = mo->symtab[i].n_value;
+			sym->size = 0; /* TODO: Is it anywhere? */
+			if (mo->symtab[i].n_type & N_EXT) {
+				// sym->type = R_BIN_MACH0_SYMBOL_TYPE_EXT;
+				sym->type = "ext";
+			} else {
+				//sym->type = R_BIN_MACH0_SYMBOL_TYPE_LOCAL;
+				sym->type = "local";
+			}
+			stridx = mo->symtab[i].n_strx;
+			if (stridx >= 0 && stridx < mo->symstrlen) {
+				symstr = (char*)mo->symstr + stridx;
+			} else {
+				symstr = "???";
+			}
+			{
```

Comment:
please extract in a separate static function.. e.g `sym->name = get_name (....);`. The need to use a new block is a clear sign we need a new function here.